File: src/SlimShader/Chunks/Rdef/ConstantBufferType.cs
using SlimShader.Util;

namespace SlimShader.Chunks.Rdef
{
	/// <summary>
	/// Indicates a constant buffer's type.
	/// Based on D3D11_CBUFFER_TYPE.
	/// </summary>
	public enum ConstantBufferType
	{
		/// <summary>
		/// A buffer containing scalar constants.
		/// </summary>
		[Description("cbuffer")]
		ConstantBuffer,

		/// <summary>
		/// A buffer containing texture data.
		/// </summary>
		TextureBuffer,

		/// <summary>
		/// A buffer containing interface pointers.
		/// </summary>
		[Description("interfaces")]
		InterfacePointers,

		/// <summary>
		/// A buffer containing binding information.
		/// </summary>
		[Description("Resource bind info for")]
		ResourceBindInformation
	}
}

File: src/cpp/Source/Chunks/Rdef/ConstantBufferType.cpp
#include "PCH.h"
#include "ConstantBufferType.h"

using namespace std;
using namespace SlimShader;

string SlimShader::ToString(ConstantBufferType value)
{
	switch (value)
	{
	case ConstantBufferType::ConstantBuffer :
		return "cbuffer";
	case ConstantBufferType::TextureBuffer :
		return "tbuffer";
	case ConstantBufferType::InterfacePointers :
		return "interfaces";
	case ConstantBufferType::ResourceBindInformation :
		return "Resource bind info for";
	default :
		throw runtime_error("Unsupported value: " + to_string((int) value));
	}
}